forte/data/data_store.py

```diff
@@ -174,72 +179,109 @@ def add_annotation_raw(
         # self.entry_dict.
         raise NotImplementedError
 
-    def set_attr(self, tid: int, attr_id: int, attr_value):
+    def set_attr(self, tid: int, attr_id: int, attr_value: Any):
         r"""This function locates the entry data with `tid` and sets its
-        attribute `attr_id` with value `attr_value`.
+        attribute `attr_id` with value `attr_value`. Called by `set_attribute()`.
 
         Args:
             tid (int): Unique id of the entry.
-            attr_id (int): id of the attribute.
-            attr_value: value of the attribute.
-
+            attr_id (int): Id of the attribute.
+            attr_value (any): Value of the attribute.
         """
-        # We retrieve the entry data from entry_dict using tid. We get its
-        # entry type. We then locate the attribute using attribute if, and update the attribute.
-
-        raise NotImplementedError
+        # We retrieve the entry data from `entry_dict` using tid.
+        # We locate the attribute using `attr_id` and update the attribute.
+        entry = self.entry_dict[tid]
+        entry[attr_id] = attr_value
 
     def get_attr(self, tid: int, attr_id: int):
         r"""This function locates the entry data with `tid` and gets the value
-        of `attr_id` of this entry.
+        of `attr_id` of this entry. Called by `get_attribute()`.
 
         Args:
             tid (int): Unique id of the entry.
-            attr_id (int): id of the attribute.
+            attr_id (int): Id of the attribute.
 
         Returns:
             The value of `attr_id` for the entry with `tid`.
         """
-        # We retrieve the entry data from entry_dict using tid. We get its
-        # entry type. We then locate the attribute using attr_id, and get the attribute.
+        # We retrieve the entry data from `entry_dict` using tid.
+        # We locate the attribute using `attr_id` and get the attribute.
+        entry = self.entry_dict[tid]
+        return entry[attr_id]
 
-        raise NotImplementedError
+    def set_attribute(self, tid: int, attr_name: str, attr_value: Any):
```

Comment:
looks like the LRU_Cache technique used by qinx would help here, maybe ask about that?

Reply:
Sure. The doc says the LRU cache should only be used when you want to reuse previously computed values. I am not sure when lru could help. When user calls get_attr multiple times on the same entry?

Reply:
yeah this is a cache for recent values. it might help a bit, you can discuss with qinx on this